### DISCUSSION TITLE
onedrive --monitor exits with a mdoerate monitor interval

### DISCUSSION THREAD

**Discussion by nbruin** [abraunegg/onedrive#1815] [Q&A] [answered]
This report concerns Fedora Core 35, standard onedrive rpm package:

    $ cat /etc/redhat-release 
    Fedora release 35 (Thirty Five)
    $ onedrive --version
    onedrive v2.4.14
    $ which onedrive
    /usr/bin/onedrive

If I issue

    $ onedrive --monitor --verbose --verbose --debug-https --monitor-interval 122

it does one sync properly and then, after about two minutes, the process produces the output below and exits with a non-error condition, 

        [DEBUG] ################################################## NEW LOOP ##################################################
        [DEBUG] fullScanCounter =                    2
        [DEBUG] syncListConfigured =                 false
        [DEBUG] fullScanRequired =                   false
        [DEBUG] syncListConfiguredFullScanOverride = false
        Starting a sync with OneDrive
        [DEBUG] performSync API scan triggers
        [DEBUG] -----------------------------
        [DEBUG] fullScanRequired =                   false
        [DEBUG] syncListConfiguredFullScanOverride = false
        [DEBUG] -----------------------------
        [DEBUG] ################################################## NEW SYNC ##################################################
        [DEBUG] Initial Scan: Call OneDrive Delta API for delta changes as compared to last successful sync.
        [DEBUG] Calling sync.applyDifferences(false);
        Applying changes of Path ID: [REMOVED]
        [DEBUG] Request URL = https://graph.microsoft.com/v1.0/drives/[REMOVED]/?select=quota
        * Too old connection (122 seconds), disconnect it
        * Connection 1 seems to be dead!
        * Closing connection 1
        * Too old connection (124 seconds), disconnect it
        * Connection 0 seems to be dead!
        * Closing connection 0

with exit code:

        $ echo $?
        141

With a monitor-interval of 120 or less, things work properly as far as I can tell. With longer intervals, including the 300 that is the default, I see the same behaviour.

As far as I can tell, these are probably Curl giving up on the too-old connections. Is that indeed the case? In that case, this would affect anyone with vanilla FC 35 settings. Should I just use a monitor interval of 120 or less?

It seems that systemd does not pick up the 141 error code as an erroneous termination (which I think it should be), so the default systemd service configuration doesn't restart. Perhaps restarting is undesirable if it is much more expensive than running a new sync through a continuing monitor session? So should onedrive perhaps try to reopen a connection if it turns out its old connections are closed, or perhaps exit with a code that is recognized by systemd as an error?

Thank you very much for maintaining such an immensely useful piece of software! I'll likely find myself having to depend on OneDrive and having good support for in in linux is a real life-saver!

**Comment by abraunegg** [comment, marked as answer]
@nbruin 
> onedrive v2.4.14

This is an obsolete and unsupported client version. Yes, this is the 'latest' that Fedora has - you need to email and make noise on RedHat's Bugzilla (https://bugzilla.redhat.com/buglist.cgi?quicksearch=onedrive) and get Zamir SUN to push the new package.

> As far as I can tell, these are probably Curl giving up on the too-old connections. Is that indeed the case? 

Yes - because this is an issue with libcurl, fixed by https://github.com/abraunegg/onedrive/pull/1789 - so until v2.4.16 is released, it is best you build from source directly so that you use this fix. Prior to this fix, the *assumption* was that the libcurl defaults, as set by Curl were being used ... but evidently they were not ....... why? I have no idea as last time I looked - library defaults were library defaults which were used by default and did not need explicitly setting to their default. Evidently my assumption was wrong and libcurl defaults had to be set to their default.

If a lower interval value works for you in the meantime - use the lower value - however all it means is that you system is checking more frequently OneDrive than is potentially necessary - wasting CPU cycles / power etc. This may or may not be of concern to you.

> It seems that systemd does not pick up the 141 error code as an erroneous termination

This is because libcurl does not actually exit correctly / exit with an error - it just terminates .... thats a Curl | libcurl bug. Thats a totally different problem and issue.

> Thank you very much for maintaining such an immensely useful piece of software! I'll likely find myself having to depend on OneDrive and having good support for in in linux is a real life-saver!

Thanks for the feedback!

**Reply by nbruin** [reply]
Wow, that was quick. Thanks for the clarification!
